### ISSUE TITLE
Incorrect parsing for contextual keyword `async`

### ISSUE THREAD

**Issue by svr93** [typescript-eslint/typescript-eslint#678]
**What code were you trying to parse?**

```ts
import { TestBed, async } from '@angular/core/testing';
```

**What did you expect to happen?**
token type `Identifier` with value `async`

**What actually happened?**
token type `Keyword` with value `async`

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `1.11.0` |
| `TypeScript`                           | `3.4.5` |
| `node`                                 | `10.14.2` |
| `npm`                                  | `6.4.1` |

**Comment by bradzacher** [comment]
According to astexplorer that isn't what comes out?

This is the result of the parse (yes, ASTExplorer is on `v1.4.2`, but we haven't changed anything in the parser)
https://astexplorer.net/#/gist/6bc10f2361963d312bd8ed3e0c46b15e/f2fdc226b280857558b5a9d90c786364c76a4ebb

<details>
<summary>This is the output of parsing against master (without location info), which is against `v1.11.0`</summary>

```js
{
  "body": [
    {
      "source": {
        "raw": "'@angular/core/testing'",
        "type": "Literal",
        "value": "@angular/core/testing",
      },
      "specifiers": [
        {
          "imported": {
            "name": "TestBed",
            "type": "Identifier",
          },
          "local": {
            "name": "TestBed",
            "type": "Identifier",
          },
          "type": "ImportSpecifier",
        },
        //////////////////////////////////
        {
          "imported": {
            "name": "async",
            "type": "Identifier",
          },
          "local": {
            "name": "async",
            "type": "Identifier",
          },
          "type": "ImportSpecifier",
        },
        //////////////////////////////////
      ],
      "type": "ImportDeclaration",
    },
  ],
  "sourceType": "module",
  "tokens": [
    {
      "type": "Keyword",
      "value": "import",
    },
    {
      "type": "Punctuator",
      "value": "{",
    },
    {
      "type": "Identifier",
      "value": "TestBed",
    },
    {
      "type": "Punctuator",
      "value": ",",
    },
    {
      "type": "Keyword",
      "value": "async",
    },
    {
      "type": "Punctuator",
      "value": "}",
    },
    {
      "type": "Identifier",
      "value": "from",
    },
    {
      "type": "String",
      "value": "'@angular/core/testing'",
    },
    {
      "type": "Punctuator",
      "value": ";",
    },
  ],
  "type": "Program",
}
```

</details>

As you can see, in both cases `async` is correctly parsed as an identifier.
The bug is not in the parser from what I can see.

**Comment by svr93** [comment]
@bradzacher Thank you for quick response!

The parser used for easy integration between TypeScript & JavaScript ecosystems. So I expect identical results with identical strings of code.
My test case:

```js
// @ts-check
const tsParser = require('@typescript-eslint/typescript-estree');
const esParser = require('esprima')

const code = `import { TestBed, async } from '@angular/core/testing';`;

const { tokens: tsTokens } = tsParser.parse(code, {
  tokens: true,
});
const esTokens = esParser.tokenize(code)

console.log(tsTokens.map(({ type, value }) => ({ type, value })))
console.log('---')
console.log(esTokens.map(({ type, value }) => ({ type, value })))
```

Result is different:
```
[ { type: 'Keyword', value: 'import' },
  { type: 'Punctuator', value: '{' },
  { type: 'Identifier', value: 'TestBed' },
  { type: 'Punctuator', value: ',' },
  { type: 'Keyword', value: 'async' }, // *?*
  { type: 'Punctuator', value: '}' },
  { type: 'Identifier', value: 'from' },
  { type: 'String', value: '\'@angular/core/testing\'' },
  { type: 'Punctuator', value: ';' } ]
---
[ { type: 'Keyword', value: 'import' },
  { type: 'Punctuator', value: '{' },
  { type: 'Identifier', value: 'TestBed' },
  { type: 'Punctuator', value: ',' },
  { type: 'Identifier', value: 'async' },
  { type: 'Punctuator', value: '}' },
  { type: 'Identifier', value: 'from' },
  { type: 'String', value: '\'@angular/core/testing\'' },
  { type: 'Punctuator', value: ';' } ]
```

Consistent results will help to use the available ESLint plugins for TypeScript code

**Comment by bradzacher** [comment]
I see - you should have included that in your initial issue!

Looking again at the example, yes - the tokens array does classify the `async` token as a keyword.

However looking at the linked issue, and the source code of the plugin, it never inspects that array, and gathers tokens via `sourceCode.getTokens(node)`.

In practice it looks like it all works as expected?

https://astexplorer.net/#/gist/8599f450e5cab7f5711b55ab02bd4a0e/bddd5558899be19236dead04055faa4348a1f66c

```
[
    {
        "type": "Keyword",
        "value": "import",
    },
    {
        "type": "Punctuator",
        "value": "{",
    },
    {
        "type": "Identifier",
        "value": "TestBed",
    },
    {
        "type": "Punctuator",
        "value": ",",
    },
    {
        "type": "Identifier",
        "value": "async",
    },
    {
        "type": "Punctuator",
        "value": "}",
    },
    {
        "type": "Identifier",
        "value": "from",
    },
    {
        "type": "String",
        "value": "'@angular/core/testing'",
    },
    {
        "type": "Punctuator",
        "value": ";",
    }
]
```

**Comment by bradzacher** [comment]
Note that I have raised #681 to fix the incorrect typing of the token.

**Comment by svr93** [comment]
@bradzacher 
I tried to clone this repo - https://github.com/lydell/eslint-plugin-simple-import-issue - and manually patch dependency with your PR changes.

Result: test is OK :)
```
diff --git a/test.js b/test.js
index 9b2896a..1139a7c 100644
--- a/test.js
+++ b/test.js
@@ -1 +1 @@
-import { TestBed, async } from '@angular/core/testing';
+import { async,TestBed } from '@angular/core/testing';
```
Plugin works as expected
